### PR TITLE
Add STM32MP13 minimal platform

### PR DESCRIPTION
**Pull request by GseoC**
This P-R adds the minimal platform of the STM32MP13.

For your interest, the STM32MP13 SoC diversity is composed by:
- STM32MP131:
      -core: 1*Cortex A7, 17*TIMERS, 5*LPTIMERS, DMA/MDMA/DMAMUX
      -storage: 2*SDMCC, 1*QSPI, FMC
      -com: USB (OHCI/EHCI, OTG), 5*I2C, 5*SPI/I2S, 8*U(S)ART
      -audio: 2*SAI
      -network: 1*ETH(GMAC)
-STM32MP133: STM32MP131 + 2*CAN, ETH2(GMAC), ADC1
-STM32MP135: STM32MP133 + DCMIPP, LTDC

This P-R aims to allow developers to perform a complete SD card boot with mainline TF-A, U-Boot and Linux kernel on this platform and to allow them to test their drivers on these components. It implements the minimal configuration for the platform to perform this boot.

### CODE REVIEW

File: core/arch/arm/dts/stm32mp13-pinctrl.dtsi [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2019 - All Rights Reserved
```
Comment (etienne-lms) [inline core/arch/arm/dts/stm32mp13-pinctrl.dtsi:3]: 2019-2022?

File: core/include/dt-bindings/regulator/st,stm32mp13-regulator.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/* SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause) */
+/*
+ * Copyright (C) 2020, STMicroelectronics - All Rights Reserved
```
Comment (etienne-lms) [inline core/include/dt-bindings/regulator/st,stm32mp13-regulator.h:3]: 2020-2022?

**Review by etienne-lms — COMMENTED** [review]
Some minor comments.

`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` for commits:
"plat-stm32mp1: support STM32MP13 gpio bank resources"
"plat-stm32mp1: conf: set variant config before common default configs"
"plat-stm32mp1: CFG_TZSRAM_START may not be define"
"drivers: clk: clk-stm32mp13: add ck_mlahb to critical clocks"
"plat-stm32mp1: tzc400: support STM32MP13"
"plat-stm32mp1: define specific STM32MP13 peripheral addresses"
"plat-stm32mp1: add APB6 memory mapping"

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` for commits:
"plat-stm32mp1: conf: introduce STM32MP1 flavorlists"
"plat-stm32mp1: conf: implement switch list for CFG_STM32MP15|13"

`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` with comments addressed for commits "plat-stm32mp1: temporary ETZPC configuration".

`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` for commit 
"drivers: stm32_gpio: temporary GPIO configuration for STM32MP13" 
with a addition in the commit log:
`Changes I2C driver to assign I2C pins to secure world for STM32MP13`.

File: core/arch/arm/plat-stm32mp1/scmi_server.c [hunk 3]
```diff
@@ -93,6 +93,55 @@ register_phys_mem(MEM_AREA_IO_NSEC, CFG_STM32MP1_SCMI_SHM_BASE,
 		.enabled = _init_enabled, \
 	}
 
+#define RESET_CELL(_scmi_id, _id, _name) \
+	[_scmi_id] = { \
+		.reset_id = _id, \
+		.name = _name, \
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/scmi_server.c:99]: Sorry, maybe out of scope: original implementation already missed these parentheses.

[the remark this answers is not in the document]
Comment (GseoC) [reply core/arch/arm/plat-stm32mp1/scmi_server.c:99]: Done, I'll use this series to update this

**Comment by GseoC** [comment]
Tags updated, thanks @etienne-lms , still pending for review:

-dts: stm32: add STM32MP13 SoCs support
-dts: stm32: add initial support of STM32MP135F-DK board
-dt-bindings: add STM32MP13 regulator bindings
-plat-stm32mp1: scmi_server update for STM32MP13
-plat-stm32mp1: conf: introduce STM32MP13 configuration
-ci: azure: add STM32MP135F-DK platform support

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` for the DT related commits:
"dts: stm32: add STM32MP13 SoCs support"
"dts: stm32: add initial support of STM32MP135F-DK board"
"dt-bindings: add STM32MP13 regulator bindings"

**Comment by etienne-lms** [comment]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` for commits:
"plat-stm32mp1: scmi_server update for STM32MP13"
"plat-stm32mp1: conf: introduce STM32MP13 configuration"
and "ci: azure: add STM32MP135F-DK platform support"

**Comment by GseoC** [comment]
Thanks for the review, tags updated, series reviewed

**Comment by jenswi-linaro** [comment]
There's quite a lot of small commits here which together adds support for the STM32MP13 platform. The commits are in my opinion too small making it harder than necessary to follow what's going on. For instance the commits:
- plat-stm32mp1: conf: introduce STM32MP1 flavorlists
- plat-stm32mp1: conf: implement switch list for CFG_STM32MP15|13
- plat-stm32mp1: conf: set variant config before common default configs
- plat-stm32mp1: CFG_TZSRAM_START may not be defined
- plat-stm32mp1: conf: introduce STM32MP13 configuration
could probably just as well gone in one commit.

Once this is merged and you look at the git history it's not super clear with which commit STM32MP13 is supported. Looking back I would probably guess "ci: azure: add STM32MP135F-DK platform support" since that's when it's added to CI. Ideally git history should be better than that. 

Since @etienne-lms now has reviewed everything I don't see any need to change the PR, but it could be worth keeping in mind the next time.

This looks safe to include in the release since it only affects other STM32MP1 flavors and I assume you've tested those.
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
Indeed I tested stm32mp1 with this series. No regression on exiting MP15 flavors. I also successfully tested that the new MP13 optee platform boots with TF-A v2.7 tag.

@GseoC, you can add my Tested-by tag on commit "ci: azure: add STM32MP135F-DK platform support":
`Tested-by: Etienne Carriere <etienne.carriere@linaro.org> (stm32mp1)`

**Comment by GseoC** [comment]
@jenswi-linaro Ok I'll keep this is mind, thanks for having a look 👍🏻 

I thought it could help to better see the steps on the platform's configuration evolution but I guess it is not that important.

Tags updated

**Comment by GseoC** [comment]
@etienne-lms At least we have the same result 😄 tag updated as well

**Pull request merged**